cocotb/triggers.py

```diff
@@ -915,9 +918,21 @@ async def with_timeout(trigger, timeout_time, timeout_unit="step"):
 
     .. versionadded:: 1.3
 
+    .. versionchanged:: 1.5
+        Support passing :term:`python:coroutine`\\ s.
+
     .. deprecated:: 1.5
         Using None as the the *timeout_unit* argument is deprecated, use ``'step'`` instead.
-   """
+    """
+
+    if inspect.iscoroutine(trigger):
+        task = cocotb.fork(trigger)
+        try:
+            res = await with_timeout(task, timeout_time, timeout_unit)
+        except cocotb.result.SimTimeoutError:
+            task.kill()
+            raise
+        return await task
```

Comment:
Isn't the `await task` unnecessary? If we get to this point, the task is already finished, and the await will just be turned into a `Join`, but that was already done by `First` and we have the result in `res`.

```suggestion
        return res
```

Reply:
Yeah, awaiting a `RunningTask` converts to a `Join`, which returns the result, not the task itself (#869).